pumpkin/src/net/packet/play.rs

```diff
@@ -894,9 +903,62 @@ impl Player {
         }
     }
 
-    pub fn handle_use_item(&self, _use_item: &SUseItem) {
-        // TODO: handle packet correctly
-        log::error!("An item was used(SUseItem), but the packet is not implemented yet");
+    // TODO: handle packet correctly
+    pub async fn handle_use_item(
+        self: &Arc<Self>,
+        _use_item: &SUseItem,
+    ) -> Result<(), Box<dyn PumpkinError>> {
+        let item_stack = *self
+            .inventory()
+            .lock()
+            .await
+            .held_item()
+            .ok_or(InventoryError::InvalidPacket)?;
+        let item = item_registry::get_item_by_id(item_stack.item_id)
+            .ok_or(InventoryError::InvalidPacket)?;
+
+        if let Some(food) = item.components.food {
+            log::debug!("Player tried eat food: {:?}", food);
+
+            let player = self.clone();
+
+            *self.eating.lock().await = Some(tokio::spawn(async move {
+                sleep(Duration::from_millis(1610)).await;
+
+                player
+                    .client
+                    .send_packet(&CEntityStatus::new(player.entity_id(), 9))
+                    .await;
+
+                if player.gamemode.load() != GameMode::Creative {
+                    let mut inventory = player.inventory().lock().await;
+                    let item_count = &mut inventory.held_item_mut().as_mut().unwrap().item_count;
+                    *item_count -= 1;
+
+                    if *item_count == 0 {
+                        *inventory.held_item_mut() = None;
+                    }
+
+                    drop(inventory);
+                }
+
+                player
+                    .set_health(
+                        player.living_entity.health.load(),
+                        player.food.load(Ordering::Relaxed) + food.nutrition,
+                        player.food_saturation.load() + food.saturation,
+                    )
+                    .await;
+
+                log::debug!("Player eated food: {:?}", food);
+            }));
```

Comment:
I see no reason to store the join handle or to spawn a new task here. What is the reason to storing the joinhandle? There is no need to spawn a new task here since we could directy handle this in this task.

Reply:
Because i can stop eating in .abort() method.
Spawning new task because i use sleep, which is blocking. Or not?

Reply:
See usages of eating field

Reply:
i can use eating.map(|h| h.abort()) in any place